xarray/tests/test_formatting.py

```diff
@@ -190,6 +193,97 @@ def test_attribute_repr(self):
         assert u'\n' not in newlines
         assert u'\t' not in tabs
 
+    def test_diff_array_repr(self):
+        da_a = xr.DataArray([[1, 2, 3], [4, 5, 7]],
+                            dims=('x', 'y'),
+                            coords={'x': ['a', 'b'], 'y': [1, 2, 3]},
+                            attrs={'units': 'm', 'description': 'desc'})
+
+        da_b = xr.DataArray([1, 2],
+                            dims='x',
+                            coords={'x': ['a', 'c'], 'label': ('x', [1, 2])},
+                            attrs={'units': 'kg'})
+
+        expected = dedent("""\
+        Left and right DataArray objects are not identical
+        Differing values:
+        L
+            array([[1, 2, 3],
+                   [4, 5, 7]])
+        R
+            array([1, 2])
+        Differing coordinates:
+        L * x        (x) <U1 'a' 'b'
+        R * x        (x) <U1 'a' 'c'
+        Left contains more coordinates:
```

Comment:
"Left contains more coordinates" sounds a little funny to me.

Maybe "Coordinates on the left DataArray but not the right" or "Coordinates only on the left object"?

Reply:
Agreed! I followed pytest's reports too blindly here.